### DISCUSSION TITLE
Consider making a smarter feed update scheduler.

### DISCUSSION THREAD

**Discussion by zoredache** [FreshRSS/FreshRSS#3862] [Ideas]
It would be nice to have a way to have a smarter feed refresh setting that adapts to the frequency of the feed. Instead of just looking at the TTL and last update.

Say you have your default to 4 hours, and one feed only updates about once every other week, it doesn't really seem like a good use of recourses, to be pulling that feed 42 times in a given week for that 1 update. I am using more bandwidth of resources of both my own, and the site I am pulling from, to get something, that probably won't exist.

Lets say you have a bunch of feeds on something like Youtube, because you the subscription page sucks. But you get rate limited because you are requesting tons of feeds frequently. Even though many of the channels only publish once a week.

Yes, I know I could probably manually adjust the refresh settings of each feed, but this seems silly. This stuff is running on a computer which should be able to easily do all the math required to be a lot smarter about picking a good refresh frequency.

I think this could be made smarter by having an  'adaptive setting', that defines a minimum, and maximum refresh frequency. Then after the first feed pull, you would pick a refresh value between the min and max based on the interval of past entries. I would imagine this interval would be selected by finding how much time elapsed between each of the last entries, and then taking the average of that. If the average is between your min and max, use that as the refresh frequency. If it is less then the min, use the min, and more then the max, use the max.

It would also be nice to have a way to rate-limit how many requests are sent to a given domain in a given time interval, so you don't get blocked in the case of youtube. So lets say you had 100 youtube feeds, only defined maximum during any given refresh cycle. This way you might be able to avoid being flagged by Google/Youtube as being abusive, and blocked.

**Comment by Alkarex** [comment]
A few pointers:

* Extension https://github.com/pe1uca/xExtension-RateLimiter
* Extension https://github.com/mgnsk/FreshRSS-AutoTTL
* https://github.com/FreshRSS/FreshRSS/issues/6556#issuecomment-2489777453

**Reply by Alkarex** [reply]
Related: feedback welcome:
* https://github.com/FreshRSS/FreshRSS/pull/7760
